addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1522,18 +1522,21 @@ const SnippetOptionWidget = Widget.extend({
      * @returns {Promise|undefined}
      */
     selectDataAttribute: function (previewMode, widgetValue, params) {
-        const dataName = params.attributeName;
-        if (dataName) {
-            if (params.saveUnit && !params.withUnit) {
-                // Values that come with an unit are saved without unit as
-                // data-attribute unless told otherwise.
-                widgetValue = widgetValue.split(params.saveUnit).join('');
-            }
-            this.$target[0].dataset[dataName] = widgetValue;
-        }
-        if (params.extraClass) {
-            this.$target.toggleClass(params.extraClass, params.defaultValue !== widgetValue);
-        }
+        this.$target[0].dataset[params.attributeName] = this._selectAttributeHelper(widgetValue, params);
```

Comment:
Split in two lines to not overflow number of characters for nothing, like you did for `selectAttribute`. Also take advantage of that change to move selectAttribute as first method to keep alphatical order.